shared/dataflow/codeql/dataflow/internal/FlowSummaryImpl.qll

```diff
@@ -1574,13 +1574,16 @@ module Make<
       /** Gets a call that targets summarized callable `sc`. */
       DataFlowCall getACall(SummarizedCallable sc);
 
+      /** Gets the enclosing callable of `source`. */
+      DataFlowCallable getSourceNodeEnclosingCallable(SourceBase source);
```

Comment:
We need to know wether the node returned by `getSourceNode` is in the same callable as `SourceBase`, and I don't think we can get that information without adding a new predicate.